### PR TITLE
feat: added stoic response

### PR DESCRIPTION
**Pull request by wendko**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 
 import (
+	"encoding/json"
 	"fmt"
+	"net/http"
 	"os"
 	"os/signal"
 	"syscall"
-
 	"github.com/bwmarrin/discordgo"
```
Comment (pickfire) [inline main.go]: Other dependencies are usually separated for a reason. I think better to not remove this.

Comment (wendko) [reply main.go]: oh thanks.. seems like space is very important in go :) will put it back

File: main.go [hunk 2]
```diff
@@ -47,4 +56,28 @@ func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
...
+
+		if err != nil {
+			fmt.Println("error retrieving stoicquotesapi", err)
+			return
+		}
+
+		defer resp.Body.Close()
+
+		var respBody StoicResponse
+		decodeErr := json.NewDecoder(resp.Body).Decode(&respBody);
```
Comment (pickfire) [inline main.go]:
```suggestion
		err = json.NewDecoder(resp.Body).Decode(&respBody)
```

Usually previously declared `err` is reused.

And the semi-colon could be removed (I haven't seen one for quiet some time).

File: main.go [hunk 3]
```diff
@@ -47,4 +56,28 @@ func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
...
+			return
+		}
+
+		defer resp.Body.Close()
+
+		var respBody StoicResponse
+		decodeErr := json.NewDecoder(resp.Body).Decode(&respBody)
+
+		if decodeErr != nil {
+			fmt.Println("error decoding stoicquotesapi response", decodeErr)
```
Comment (pickfire) [inline main.go]:
```suggestion
		err = json.NewDecoder(resp.Body).Decode(&respBody)

		if err != nil {
			fmt.Println("error decoding stoicquotesapi response", err)
```

Usually `err` is reused.

**Review by pickfire — APPROVED** [review]
Looks good to me.

File: main.go (hunk 2 above)
Comment (wendko) [reply main.go]: fixed

**Pull request merged**